### PR TITLE
Restrict `--set` & `--set-file`  options in `cylc vr`

### PR DESCRIPTION
**Pull request by wxtim**
This PR documents the bug as well as fixing it. There is no related issue. Follow-up to #5229

## Synopsis

`--set` and `--set-file` were included by mistake in `cylc vr` because I copied the general form of VR from `cylc vip` where it makes sense. Reinstall doesn't use these CLI variables so they won't change the workflow. This is potentially misleading to users.

## Code to try

```ini
#!jinja2
# Cylc Workflow
[scheduler]
    allow implicit tasks = ture
[scheduling]
    [[graph]]
        {{CYCLE}}= foo => bar => baz
```

then,

On the command line

```bash
cylc vip --set 'CYCLE="PT1H"'
...

# Does not change the running workflow!
cylc vr --set 'CYCLE="PT2H"' 
```

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] No tests, change is specification rather than tests.
- [x] No `CHANGES.md` entry - this effectively fixes a recently added PR before relase.
- [x] NO [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.

### CODE REVIEW

File: cylc/flow/scripts/validate_reinstall.py [hunk 1]
```diff
@@ -92,6 +92,31 @@ def get_option_parser() -> COP:
...
+    Template vars set by --set (options.templatevars) or --set-file
+    (optiions.templatevars_file) are only valid if the workflow is stopped
+    and vr will play it.
+
+    args:
+        is_running: Is workflow running?
+        tvars: options.tvars, from `--set`
+        tvars_file: options.tvars_file, from `--set-file`
+    """
+    if is_running and (tvars or [] or tvars_file or []):
```
Comment (oliver-sanders) [inline cylc/flow/scripts/validate_reinstall.py:109]:
```suggestion
    if is_running and (tvars or tvars_file):
```

Comment (MetRonnie) [inline cylc/flow/scripts/validate_reinstall.py:109]:
```suggestion
    if is_running and (tvars or tvars_file):
```

**Review by oliver-sanders — APPROVED** [review]
a

File: cylc/flow/scripts/validate_reinstall.py [hunk 2]
```diff
@@ -92,6 +92,31 @@ def get_option_parser() -> COP:
...
+        msg = (
+            'Template variables can only be changed if a'
+            ' workflow is stopped. You tried to change:')
+        for tvar in tvars or [] + tvars_file or []:
+            msg += f'\n * {tvar}'
+        LOG.warning(msg)
```
Comment (MetRonnie) [inline cylc/flow/scripts/validate_reinstall.py:115]: I don't think it needs to list the template variables back to the user
```suggestion
        LOG.warning(
            'Template variables (from --set/--set-file) can '
            'only be changed if the workflow is stopped.'
        )
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**